src/mon/MgrMonitor.cc

```diff
@@ -159,6 +159,10 @@ bool MgrMonitor::preprocess_beacon(MonOpRequestRef op)
     return true;
   }
 
+  if (!mon->is_leader()) {
```

Comment:
maybe a better fix is to just do the minimum check in `MgrMonitor::preprocess_beacon()`: just check the cap and fsid, and move the other checks to `prepare_beacon()`. please note that we update `last_beacon[]` twice, which does not make sense.